lte/gateway/c/oai/tasks/s1ap/s1ap_mme_handlers.c

```diff
@@ -2645,6 +2645,173 @@ int s1ap_mme_handle_handover_command(
+    return RETURNerror;
+  }
+  TBCD_TO_PLMN_T(&ie->value.choice.EUTRAN_CGI.pLMNidentity, &ecgi.plmn);
+  BIT_STRING_TO_CELL_IDENTITY(
+      &ie->value.choice.EUTRAN_CGI.cell_ID, ecgi.cell_identity);
+
+  // TAI
+  S1AP_FIND_PROTOCOLIE_BY_ID(
+      S1ap_HandoverNotifyIEs_t, ie, container, S1ap_ProtocolIE_ID_id_TAI, true);
+  if (!ie) {
+    OAILOG_ERROR(LOG_S1AP, "Incorrect IE \n");
+    return RETURNerror;
+  }
+  OCTET_STRING_TO_TAC(&ie->value.choice.TAI.tAC, tai.tac);
+  if (!(ie->value.choice.EUTRAN_CGI.pLMNidentity.size == 3)) {
+    OAILOG_ERROR(LOG_S1AP, "Incorrect PLMN size \n");
+    return RETURNerror;
+  }
+  TBCD_TO_PLMN_T(&ie->value.choice.TAI.pLMNidentity, &tai.plmn);
+
+  // imsi for logging
+  hashtable_uint64_ts_get(
+      imsi_map->mme_ue_id_imsi_htbl, (const hash_key_t) mme_ue_s1ap_id,
+      &imsi64);
+
+  // get existing UE context
+  if ((src_ue_ref_p = s1ap_state_get_ue_mmeid(mme_ue_s1ap_id)) == NULL) {
+    OAILOG_ERROR_UE(
+        LOG_S1AP, imsi64,
+        "source MME_UE_S1AP_ID (" MME_UE_S1AP_ID_FMT
+        ") does not point to any valid UE\n",
+        mme_ue_s1ap_id);
+    OAILOG_FUNC_RETURN(LOG_S1AP, RETURNerror);
+  } else {
+    // create new UE context, remove the old one.
+    new_ue_ref_p =
+        s1ap_state_get_ue_enbid(target_enb->sctp_assoc_id, tgt_enb_ue_s1ap_id);
+    if (new_ue_ref_p != NULL) {
+      OAILOG_ERROR_UE(
+          LOG_S1AP, imsi64,
+          "S1AP:Handover Notify- Recieved ENB_UE_S1AP_ID is not Unique "
+          "Drop Handover Notify for eNBUeS1APId:" ENB_UE_S1AP_ID_FMT "\n",
+          tgt_enb_ue_s1ap_id);
+      OAILOG_FUNC_RETURN(LOG_S1AP, RETURNerror);
+    }
+    if ((new_ue_ref_p = s1ap_new_ue(state, assoc_id, tgt_enb_ue_s1ap_id)) ==
+        NULL) {
+      // If we failed to allocate a new UE return -1
+      OAILOG_ERROR_UE(
+          LOG_S1AP, imsi64,
+          "S1AP:Handover Notify- Failed to allocate S1AP UE Context, "
+          "eNBUeS1APId:" ENB_UE_S1AP_ID_FMT "\n",
+          tgt_enb_ue_s1ap_id);
+      OAILOG_FUNC_RETURN(LOG_S1AP, RETURNerror);
+    }
+    new_ue_ref_p->s1_ue_state    = S1AP_UE_CONNECTED;  // handover has completed
+    new_ue_ref_p->enb_ue_s1ap_id = tgt_enb_ue_s1ap_id;
+    // Will be allocated by NAS
+    new_ue_ref_p->mme_ue_s1ap_id = mme_ue_s1ap_id;
+
+    new_ue_ref_p->s1ap_ue_context_rel_timer.id =
+        src_ue_ref_p->s1ap_ue_context_rel_timer.id;
+    new_ue_ref_p->s1ap_ue_context_rel_timer.sec =
+        src_ue_ref_p->s1ap_ue_context_rel_timer.sec;
+    new_ue_ref_p->sctp_stream_recv =
+        src_ue_ref_p->s1ap_handover_state.target_sctp_stream_recv;
+    new_ue_ref_p->sctp_stream_send =
+        src_ue_ref_p->s1ap_handover_state.target_sctp_stream_send;
+
+    // generate a message to update bearers
+    s1ap_mme_itti_s1ap_handover_notify(
+        mme_ue_s1ap_id, src_ue_ref_p->s1ap_handover_state, tgt_enb_ue_s1ap_id,
+        src_ue_ref_p->s1ap_handover_state.target_enb_id, assoc_id, ecgi,
```

Comment:
we are already passing the `s1ap_handover_state`, why do we need to pass `s1ap_handover_state.target_enb_id` additionally?

Reply:
Good point!